source/mesh_deformation/fastscape.cc

```diff
@@ -0,0 +1,1237 @@
+                          else if (j==(nx-1))
+                            slope = bottom_flux/kdd - std::tan(slopep[index_bot+nx-1]*numbers::PI/180.);
+                          else
+                            slope = bottom_flux/kdd - std::tan(slopep[index_bot+nx]*numbers::PI/180.);
+
+                          h[index_bot] = h[index_bot] + slope*2*dx;
+                        }
+
+                      if (bottom == 0 && top == 0)
+                        {
+                          int side = index_bot;
+                          int op_side = index_top;
+                          int jj = nx;
+
+                          if (vy[index_bot+nx-1] > 0 && vy[index_top-nx-1] >= 0)
+                            {
+                              side = index_top;
+                              op_side = index_bot;
+                              jj = -nx;
+                            }
+                          else if (vy[index_bot+nx-1] <= 0 && vy[index_top-nx-1] < 0)
+                            {
+                              side = index_bot;
+                              op_side = index_top;
+                              jj = nx;
+                            }
+                          else
+                            continue;
+
+                          // Set top ghost node
+                          h[index_top] = h[side+jj];
+                          vx[index_top] = vx[side+jj];
+                          vy[index_top] = vy[side+jj];
+                          vz[index_top] = vz[side+jj];
+
+                          // Set bottom ghost node
+                          h[index_bot] = h[side+jj];
+                          vx[index_bot] = vx[side+jj];
+                          vy[index_bot] = vy[side+jj];
+                          vz[index_bot] = vz[side+jj];
+
+                          // Set opposing ASPECT boundary so it's periodic.
+                          h[op_side-jj] = h[side+jj];
+                          vx[op_side-jj] = vx[side+jj];
+                          vz[op_side-jj] = vz[side+jj];
+                          vy[op_side-jj] = vy[side+jj];
+                        }
+                    }    
+    }
+
+    // TODO: Give better explanations of variables and cite the fastscape documentation.
+    template <int dim>
+    void FastScape<dim>::declare_parameters(ParameterHandler &prm)
+    {
+      prm.enter_subsection ("Mesh deformation");
+      {
+        prm.enter_subsection ("Fastscape");
+        {
+          prm.declare_entry("Number of steps", "10",
+                            Patterns::Integer(),
+                            "Number of steps per ASPECT timestep");
+          prm.declare_entry("Maximum timestep", "10e3",
+                            Patterns::Double(0),
+                            "Maximum timestep for FastScape.");
+          prm.declare_entry("Vertical exaggeration", "-1",
+                            Patterns::Double(),
```

Comment:
or ints?

Reply:
For the vertical exaggeration, FastScape expects a double value.